Tests/UnitTests/Identity/ProductPaidPriceTests.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ProductPaidPriceTests.swift
+//
+//  Created by Facundo Menzella on 15/1/25.
+
+import Foundation
+import Nimble
+import StoreKit
+import XCTest
+
+@testable import RevenueCat
+
+@available(iOS 11.2, macOS 10.13.2, watchOS 6.2, tvOS 11.2, *)
+final class ProductPaidPriceTests: TestCase {
+
+    // MARK: - Basic Initialization Tests
+
+    func testBasicInitialization() {
+        let price = ProductPaidPrice(currency: "USD", amount: 4.99, formatted: "$4.99")
+
+        expect(price.currency) == "USD"
+        expect(price.amount) == 4.99
+        expect(price.formatted) == "$4.99"
+    }
+
+    // MARK: - Convenience Initializer Tests
+
+    func testConvenienceInitializerFormatsUSDCorrectly() {
+        let usLocale = Locale(identifier: "en_US")
+        let price = ProductPaidPrice(currency: "USD", amount: 4.99, locale: usLocale)
+
+        expect(price.currency) == "USD"
+        expect(price.amount) == 4.99
+        expect(price.formatted) == "$4.99"
+    }
+
+    func testConvenienceInitializerFormatsEURCorrectly() {
+        let frenchLocale = Locale(identifier: "fr_FR")
+        let price = ProductPaidPrice(currency: "EUR", amount: 7.99, locale: frenchLocale)
+
+        expect(price.currency) == "EUR"
+        expect(price.amount) == 7.99
+        expect(price.formatted).to(contain("7,99"))
+        expect(price.formatted).to(contain("€"))
+    }
+
+    // MARK: - Static formatPrice Method Tests
+
+    func testFormatPriceWithUSD() {
+        let usLocale = Locale(identifier: "en_US")
+        let formatted = ProductPaidPrice.formatPrice(amount: 9.99, currency: "USD", locale: usLocale)
+
+        expect(formatted) == "$9.99"
+    }
+
+    func testFormatPriceWithEUR() {
```

Comment:
Nitpick but should we also add a test where the locale does not match the currency? For example, Spain locale for USD?